src/parser/jobs/ast/modules/EarthlyStar.tsx

```diff
@@ -50,20 +53,33 @@ export default class EarthlyStar extends Module {
 	}
 
 	private onPlace(event: CastEvent) {
+		this.uses++
+
 		// this was prepull
 		if (event.timestamp < this.parser.fight.start_time) {
-			this.debug('prepull')
+			this.prepull = true
 		}
 
 		// TODO: Instead determine how far back they used it prepull by checking explosion time.
 		if (this.lastUse === 0) {
 			this.lastUse = this.parser.fight.start_time
 		}
 
-		const held = event.timestamp - this.lastUse - (this.data.actions.EARTHLY_STAR.cooldown * 1000)
-		if (held > 0) {
-			this.totalHeld += held
+		let drift = 0
+
+		if (this.uses === 1 && !this.prepull) {
+			// The first use, take holding as from the start of the fight
+			drift = event.timestamp - this.parser.fight.start_time
+		} else {
+			// Take holding as from the time it comes off cooldown
+			drift = event.timestamp - this.lastUse - (this.data.actions.EARTHLY_STAR.cooldown * 1000)
+		}
+
+		// Keep track of total drift time not using star
+		if (drift > 0) {
+			this.totalHeld += drift
 		}
```

Comment:
This is generally messy and could be done more simply with a clamp. In the interest of getting it through, ignoring, but at some point drift and also held times should probably be rethought. Could just initialise lastUse to be fight start time.

Reply:
Easy enough to change